### PR TITLE
add ZManaged#allocated

### PR DESCRIPTION
**Pull request by mschuwalow**
Maybe we can find a better name for this.
My vote still goes to ZManaged#dunno

/cc @iravid @pshemass

### CODE REVIEW

File: core/shared/src/main/scala/zio/ZManaged.scala [hunk 1]
```diff
@@ -143,6 +138,18 @@ final class ZManaged[-R, +E, +A] private (reservation: ZIO[R, E, Reservation[R,
   ): ZManaged[R1, E1, B] =
     ZManaged.absolve(ev(self))
 
+  /**
+   * Preallocates the managed resource, resulting in a ZManaged that reserves and acquires immediately and cannot fail.
+   */
+  final def allocated: ZIO[R, E, Managed[Nothing, A]] =
```
Comment (pshemass) [inline core/shared/src/main/scala/zio/ZManaged.scala]: just for my curiosity what happen with ```E``` in this case? is it swallowed?

**Comment by pshemass** [comment]
so let go back to the name :)

for me ```allocated``` doesn't sound bad. it could be also ```acquired``` or  ```reserved```.

/cc: @iravid @mijicd

**Comment by mijicd** [comment]
+1 for preallocate.

**Comment by pshemass** [comment]
```preallocate``` let it be :)

**Comment by mschuwalow** [comment]
Cool, I'll update the pr once I get home

**Pull request merged**